integrations/test-gitlab-v2/client.py

```diff
@@ -0,0 +1,105 @@
+import httpx
+from typing import List, Dict, Any, AsyncIterator
+from loguru import logger
+from port_ocean.context.ocean import ocean
+
+class GitLabClient:
+    def __init__(self, base_url: str, token: str):
+        self.base_url = base_url
+        self.token = token
+        self.headers = {"Authorization": f"Bearer {self.token}"}
+        logger.info(f"GitLabClient initialized with base_url: {self.base_url}")
+
+    async def _request(self, method: str, endpoint: str, **kwargs) -> dict:
+        url = f"{self.base_url}/{endpoint}"
+        logger.info(f"Making {method} request to {url}")
+        async with httpx.AsyncClient() as client:
+            response = await client.request(method, url, headers=self.headers, timeout=60, **kwargs)
```

Comment:
let's not create our own async client. use the one from port_ocean 

from port_ocean.utils import http_async_client